### PR TITLE
DS-367 Implement Tooltips

### PR DESCRIPTION
**Pull request by jeremy-davis-sonarsource**

### CODE REVIEW

File: src/components/tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  );
+}
+
+Tooltip.displayName = 'Tooltip';
+
+const TooltipContent = styled(radixTooltip.Content)`
+  border-radius: var(--echoes-border-radius-200);
+  padding: var(--echoes-dimension-space-50) var(--echoes-dimension-space-150);
+  color: white;
+  background-color: var(--echoes-color-background-inverse);
```
Comment (gregaubert) [inline src/components/tooltip/Tooltip.tsx:84]: We also need to set the default typography and to use a design-token the font-color to make it work correctly with dark theme

File: src/components/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  content?: React.ReactNode;
+  isOpen?: boolean;
+  side?: TooltipSide;
+}
+
+/* This is the distance between the point of the arrow and the trigger */
+const TOOLTIP_OFFSET = 4;
+
+/* This is the padding between the edge of the tooltip and the arrow. */
+const ARROW_PADDING = 12;
```
[the remark this answers is not in the document]
Comment (jeremy-davis-sonarsource) [reply src/components/tooltip/Tooltip.tsx:49]: Short answer: no
Long answer: These are extremely specific. I'm not sure it makes sense to have them as design tokens (maybe the tooltip offset, but then we'd need a dedicated token to handle the format: `number` (4) vs `string` ("0.25rem")

File: src/components/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  return (
+    <radixTooltip.Root open={isOpen}>
+      <radixTooltip.Trigger asChild>{children}</radixTooltip.Trigger>
+      <radixTooltip.Portal>
+        <TooltipContent
+          align={align}
+          arrowPadding={ARROW_PADDING}
+          side={side}
```
[the remark this answers is not in the document]
Comment (jeremy-davis-sonarsource) [reply src/components/tooltip/Tooltip.tsx:68]: The default is 'top' and I think that's the standard (or the most usual). I think it makes sense to defer to radix

File: src/components/tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  if (!isDefined(content) || (typeof content === 'string' && content.length === 0)) {
+    return <>{children}</>;
+  }
+
+  return (
+    <radixTooltip.Root open={isOpen}>
+      <radixTooltip.Trigger asChild>{children}</radixTooltip.Trigger>
+      <radixTooltip.Portal>
+        <TooltipContent
+          align={align}
```
[the remark this answers is not in the document]
Comment (jeremy-davis-sonarsource) [reply src/components/tooltip/Tooltip.tsx:66]: the default is 'center'

File: src/components/tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+Tooltip.displayName = 'Tooltip';
+
+const TooltipContent = styled(radixTooltip.Content)`
+  border-radius: var(--echoes-border-radius-200);
+  padding: var(--echoes-dimension-space-50) var(--echoes-dimension-space-150);
+  color: white;
+  background-color: var(--echoes-color-background-inverse);
+  box-shadow: var(--echoes-box-shadow-medium);
+  max-width: 400px;
```
[the remark this answers is not in the document]
Comment (jeremy-davis-sonarsource) [reply src/components/tooltip/Tooltip.tsx:86]: You're right! Not sure why he didn't use it in figma

File: src/components/tooltip/TooltipProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as radixTooltip from '@radix-ui/react-tooltip';
+import { PropsWithChildren } from 'react';
+
+export function TooltipProvider({ children }: PropsWithChildren<{}>) {
```
[the remark this answers is not in the document]
Comment (jeremy-davis-sonarsource) [reply src/components/tooltip/TooltipProvider.tsx:24]: There's only one we provide for now, so let's wait for others before providing a pyramid.
(react-intl is a peerDependency, so I don't think it's a good idea to include it in our own pyramid, I expect consumer apps to already have it in place)

File: src/components/tooltip/Tooltip.tsx (hunk 1 above)
Comment (jeremy-davis-sonarsource) [reply src/components/tooltip/Tooltip.tsx:84]: How did I miss that :facepalm:

File: src/components/tooltip/Tooltip.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gregaubert) [reply src/components/tooltip/Tooltip.tsx:49]: Yeah make sense, I'm just thinking these things are design decisions, so if the designer decide to change them he could update the design-token and then the change would be "automatically" reflected in the lib.

**Review by gregaubert — APPROVED** [review]
Looks good! Some other fixes coming up with the other PR. Let's merge this one already!

**Pull request merged**